src/Workspaces/Core/Portable/Remote/DefaultRemoteHostClientServiceFactory.RemoteHostClientService.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    internal partial class DefaultRemoteHostClientServiceFactory
+    {
+        public class RemoteHostClientService : IRemoteHostClientService
+        {
+            private readonly Task<RemoteHostClient> _instanceTask;
+
+            public RemoteHostClientService(Workspace workspace)
+            {
+                var remoteHostClientFactory = workspace.Services.GetService<IRemoteHostClientFactory>();
+                if (remoteHostClientFactory == null)
+                {
+                    // no implementation of remote host client
+                    return;
+                }
+
+                _instanceTask = Task.Run(() => remoteHostClientFactory.CreateAsync(workspace, CancellationToken.None), CancellationToken.None);
```

Comment:
> it jumps only when IRemoteHostClientFactory exist. which doesn't exist in default layer.

Ok.  So it does need to do something expensive when the factory exists.  So it seems like this should be cancellable in some manner.